### ISSUE TITLE
[bug] Lots of audio dropping out

### ISSUE THREAD

**Issue by sayangel** [Placeholder-Software/Dissonance#106]
## Context
We're testing version 6.2.3 and getting a lot of the following errors: 

```
[Warning] [Dissonance:Recording] (12:04:01.300) BasePreprocessingPipeline: Lost 480 samples in the preprocessor (buffer full), injecting silence to compensate
Dissonance.Audio.Capture.BasePreprocessingPipeline.Dissonance.Audio.Capture.IMicrophoneSubscriber.ReceiveMicrophoneData()    Assets/Plugins/Dissonance/Core/Audio/Capture/BasePreprocessingPipeline.cs:175
```

This results in voice coming in and out intermittently.

## Expected Behavior
Audio should not drop out.

## Actual Behavior
Audio drops in and out and the error above is shown in the editor whenever it happens.

## Your Environment
_Include as many relevant details about the environment you experienced the bug in_

- **Dissonance version used**: v6.2.3

- **Unity version**:  5.6.4p2

- **Editor Operating System and version**: Windows 10

- **Build Settings**: Standalone Windows

**Comment by martindevans** [comment]
Is this a regresssion? If so what version of Dissonance were you using before?

Could you raise the logging level to Debug (Window > Dissonance > Diagnostic Settings) and send me the log.

**Comment by haydenjameslee** [comment]
(Another team member here) Yes it is a regression. Our main branch is on 6.0.2 and doesn't ever have this problem.

**Comment by martindevans** [comment]
Well that's rather concerning!

I'm just about to put out the next version to the store, could one of you email me and I'll send you a copy of that. If the issue persists I'll have to investigate in more detail.

**Comment by haydenjameslee** [comment]
Emailed.

**Comment by martindevans** [comment]
I've sent you back the Unity 5.6 package of `Dissonance 6.2.4`. As you'll see from the release notes this version includes a number of bugfixes, including some to the preprocessor which is where you're having problems.

If it doesn't fix your problem  please send me the log I mentioned before and I'll get right on investigating it :)

**Comment by haydenjameslee** [comment]
We tested 6.2.4 today and didn't notice the bug happening at all. There were still some logs but I've realized we had the Dissonance log severity set to "errors" in our main branch so that's potentially why we haven't seen this warning before.

**Comment by martindevans** [comment]
Excellent, It sounds like that's been resolved by the preprocessor fix in `6.2.4`. I'll close this issue now, feel free to re-open it if the it becomes a problem again.

**Comment by sayangel** [comment]
Thanks as always @martindevans !
